### ISSUE TITLE
softmax_with_cross_entropy API在软标签下label是否需要归一化

### ISSUE THREAD

**Issue by gqzhang-ai** [PaddlePaddle/Paddle#63678]
### bug描述 Describe the Bug

[paddle.nn.functional. softmax_with_cross_entropy](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/softmax_with_cross_entropy_cn.html#softmax-with-cross-entropy) API：
在软标签下，文档中标明（ 每个样本以一定的概率被分配至多个类别中，概率和为 1），是否可以理解为label需要经过Normalize，如果是的话，代码中却没有对label是否经过归一化进行检查，可以测试下述代码：
`import paddle
paddle.set_device("cpu")

logits = np.random.uniform(low=0, high=1, size=(1, 5, 5, 5)).astype(np.float32)
labels = np.random.uniform(low=0, high=1, size=(1, 5, 5, 5)).astype(np.float32)

logits = paddle.to_tensor(logits, stop_gradient=False)
labels = paddle.to_tensor(labels, stop_gradient=False)

out = paddle.nn.functional.softmax_with_cross_entropy(logits=logits, label=labels, soft_label=True)

print(out)`

**Comment by zzjjay** [comment]
在软标签下，计算 cross entropy 需要对label进行归一化。

paddle.nn.functional.softmax_with_cross_entropy 仅会对 logits 做 softmax 归一化。

可以先使用 paddle.sparse.nn.functional.softmax 对软标签 label进行归一化，再传入 softmax_with_cross_entropy 计算 cross entropy

**Comment by gqzhang-ai** [comment]
那是否可以在 softmax_with_cross_entropy 代码中做check，判断用户在使用软标签时，label是否进行了归一化，从而避免用户使用错误，以便提前暴露问题，方便排查。

**Comment by zzjjay** [comment]
在软标签模式下，传入的label是样本在不同类别的概率分布，这个是软标签的天然含义，因此对label做归一化并不是该api所需要承担的职责。
